cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java

```diff
@@ -82,6 +84,8 @@
  *
  * Lookup of schedules by trigger key is by first finding the all triggers for that event key (using the index),
  * then mapping each of these triggers to the schedule it belongs to.
+ *
+ * Note that deleting a schedule will leave an empty row with a deletion timestamp in the table.
```

Comment:
Better to add a comment on why we leave the timestamp in the table for deletion instead of removing the row.

Reply:
good point. will add that.